wlct/tournaments.py

```diff
@@ -3919,6 +3919,47 @@ def create_season(self, season_name, games_at_once):
         season = PromotionalRelegationLeagueSeason(pr_tournament=self, name=season_name, created_by=self.created_by, private=True, games_at_once=int(games_at_once))
         season.save()
 
+    def copy_season(self, season_name, copy_season_id):
+        # Get original season to copy
+        orig_season = PromotionalRelegationLeagueSeason.objects.filter(id=int(copy_season_id))
+        # Ensure parameters are correct before copying
+        if  not orig_season:
+            raise ValueError("Seasonal with id {} could not be found".format(copy_season_id))
+        # Get the # of games and template from original season
+        games_at_once = orig_season[0].games_at_once
+        template = orig_season[0].season_template;
+        if len(season_name) < 3 or len(season_name) > 251:
+            raise ValueError("Season name must be between 3-250 characters.")
+        if games_at_once < 1 or games_at_once > 100:
+            raise ValueError("Games at once must be 1-100 inclusive.")
+        # Create a new season given original season
+        season = PromotionalRelegationLeagueSeason(pr_tournament=self, name=season_name, created_by=self.created_by, private=True, games_at_once=games_at_once, season_template=template)
```

Comment:
Can use the pk trick here by setting orig_season.pk = None, orig_season.name = season_name, then save().